testsuite/MDAnalysisTests/core/test_groups.py

```diff
@@ -588,13 +588,13 @@ def test_groupby_float(self, u):
             assert len(g) == 25
 
     def test_groupby_string(self, u):
-        gb = u.atoms.groupby('types')
+        gb = u.atoms.groupby('segids')
```

Comment:
you can parametrize this test on string types.

```python
@pytest.mark.parametrize('string', ['segids', b'segid', u'segids'])
def test_groupby_string(self, u, string):
    gb = u.atoms.groupby(string)
    ...
```

Reply:
@kain88-de thhe test on string types fails for b'segids' on python3 with a 

```
 TypeError: getattr(): attribute name must be string
```
Using the decode method on toppattrs in the groupby function would solve this issue. Which encoding should I use: ascii or utf-8 ?